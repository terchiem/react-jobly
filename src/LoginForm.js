import React, { useState } from 'react';
import './Form.css';

/** Form component for a user to sign in. */

function LoginForm({ logIn }) {
  const [formData, setFormData] = useState({
    username: "",
    password: ""
  });


  function handleChange(evt) {
    const { name, value } = evt.target;
    setFormData(data => ({...data, [name]: value}));
  }

  return (
    <form className="Form" onSubmit={logIn}>
      <div className="Form-group">
        <label htmlFor="username" className="Form-label">Username</label>
        <input 
          className="Form-input"
          id="username" 
          name="username" 
          value={formData.username} 
          onChange={handleChange} 
        />
      </div>
      <div className="Form-group">
        <label htmlFor="password" className="Form-label">Password</label>
        <input 
          className="Form-input"
          id="password" 
          name="password" 
          type="password"
          value={formData.password} 
          onChange={handleChange} 
        />
      </div>
      <button className="Form-submit">Submit</button>
    </form>
  )
}

export default LoginForm;