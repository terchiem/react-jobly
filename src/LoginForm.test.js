import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import LoginForm from './LoginForm';

test('renders without crashing', () => {
  render(
    <MemoryRouter>
      <LoginForm />
    </MemoryRouter>
  );
});

test('matches snapshot', () => {
  const { asFragment } = render(
    <MemoryRouter>
      <LoginForm />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});