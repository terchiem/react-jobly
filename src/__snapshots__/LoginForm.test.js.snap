// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="Form"
  >
    <div
      class="Form-group"
    >
      <label
        class="Form-label"
        for="username"
      >
        Username
      </label>
      <input
        class="Form-input"
        id="username"
        name="username"
        value=""
      />
    </div>
    <div
      class="Form-group"
    >
      <label
        class="Form-label"
        for="password"
      >
        Password
      </label>
      <input
        class="Form-input"
        id="password"
        name="password"
        type="password"
        value=""
      />
    </div>
    <button
      class="Form-submit"
    >
      Submit
    </button>
  </form>
</DocumentFragment>
`;
